shared/common-adapters/with-tooltip.native.js

```diff
@@ -1,7 +1,146 @@
 
-// Tooltip is not supported on mobile.
-export default (props: Props) => <View style={props.containerStyle}>{props.children}</View>
+// This uses a similar mechanism to relative-popup-hoc.desktop.js. It's only
+// ever used for tooltips on mobile for now. If we end up needing relative
+// positioning in other places, we should probably make a
+// relative-popup-hoc.native.js. Also, this only supports "bottom center" and
+// "top center" for now.
+
+const Kb = {
+  Animated,
+  Box,
+  ClickableBox,
+  FloatingBox,
+  Text,
+}
+
+type State = {
+  left: number,
+  top: number,
+  visible: boolean,
+}
+
+const measureCb = resolve => (_1, _2, width, height, left, top) => resolve({height, left, top, width})
+
+class WithTooltip extends React.PureComponent<PropsWithTimer<Props>, State> {
+  state = {
+    left: 0,
+    top: 0,
+    visible: false,
+  }
+  _clickableRef = React.createRef()
+  _tooltipRef = React.createRef()
+  _onClick = () => {
+    if (!this._clickableRef.current || !this._tooltipRef.current || this.state.visible) {
+      return
+    }
+
+    const screenWidth = Dimensions.get('window').width
+    const screenHeight = Dimensions.get('window').height
+
+    Promise.all([
+      new Promise(
+        resolve => this._clickableRef.current && this._clickableRef.current.measure(measureCb(resolve))
+      ),
+      new Promise(
+        resolve => this._tooltipRef.current && this._tooltipRef.current.measure(measureCb(resolve))
+      ),
+    ]).then(([c, t]) => {
+      if (!this._mounted) {
+        return
+      }
+
+      const constrainLeft = ideal => Math.max(0, Math.min(ideal, screenWidth - t.width))
+      const constrainTop = ideal => Math.max(0, Math.min(ideal, screenHeight - t.height))
+      this.props.position === 'bottom center'
+        ? this.setState({
+            left: constrainLeft(c.left + c.width / 2 - t.width / 2),
+            top: constrainTop(c.top + c.height),
+            visible: true,
+          })
+        : this.setState({
+            left: constrainLeft(c.left + c.width / 2 - t.width / 2),
+            top: constrainTop(c.top - t.height),
+            visible: true,
+          }) // default to top center
+
+      this.props.setTimeout(() => this.setState({visible: false}), 3000)
+    })
+  }
+
+  _mounted = false
+  componentDidMount() {
+    this._mounted = true
+  }
+  componentWillUnmount() {
+    this._mounted = false
+  }
+
+  render() {
+    if (!this.props.showOnPressMobile) {
+      return <Kb.Box style={this.props.containerStyle}>{this.props.children}</Kb.Box>
+    }
+
+    return (
+      <>
+        <Kb.Box
+          style={this.props.containerStyle}
+          ref={
+            // $FlowIssue
+            this._clickableRef
+          }
+          className={this.props.className}
+        >
+          <Kb.ClickableBox onClick={this._onClick}>{this.props.children}</Kb.ClickableBox>
+        </Kb.Box>
+        <Kb.Animated from={{}} to={{opacity: this.state.visible ? 1 : 0}}>
```

Comment:
I think this or one if its children should have `pointerEvents="none"`. You can't scroll / tap on anything that's below the tooltip area
![tooltip](https://user-images.githubusercontent.com/11968340/55988881-049e7a00-5c73-11e9-8614-fd36a90d833d.gif)